src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -130,17 +180,39 @@ public static Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(st
             return CreateFromResourceResponseAsync(resourceUri, default);
         }
 
+        /// <summary>
+        /// Create a list of the known authenticate parameters by attempting to call the resource unauthenticated, and analyzing the response.
+        /// </summary>
+        /// <param name="resourceUri">URI of the resource.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+        /// <returns>a list of WWW-Authenticate Parameters extracted from a response to the unauthenticated call.</returns>
+        public static async Task<IEnumerable<WwwAuthenticateParameters>> CreateKnownParametersFromResourceResponseAsync(string resourceUri, CancellationToken cancellationToken = default)
```

Comment:
I recommend against calling the RP like this. We will almost certainly be asked to allow injection of HttpClient etc. So this is not the place for this sort of functionality.

Can't we just have a `CreateFromResponseHeaders(HttpResponse)` which returns a dictionary of params, [scheme] -> [WwwAuthenticateParameters] for all known schemes? Or maybe even all schemes.